assets/js/laws-regs-parser.js

```diff
@@ -242,3 +235,157 @@ function shareLink(e, divType) {
 }
+
+function addCopyEventListeners(btn, divType) {
+  btn.addEventListener('click', (e) => {
+    copyText(e, divType);
+  });
+  btn.addEventListener('mouseover', (e) => {
+    highlightText(e, divType);
+  });
+  btn.addEventListener('mouseleave', (e) => {
+    unHighlightText(e, divType);
+  });
+}
+
+function addPrintEventListeners(btn, divType) {
+  btn.addEventListener('click', (e) => {
+    printText(e, divType);
+  });
+  btn.addEventListener('mouseover', (e) => {
+    highlightText(e, divType);
+  });
+  btn.addEventListener('mouseleave', (e) => {
+    unHighlightText(e, divType);
+  });
+}
+
+function navResults(e, dir, prevBtn, nextBtn, currentCount, totalCount) {
+  e.preventDefault();
+  const count = parseInt(currentCount.innerText);
+  if (dir == 'next') {
+    if (count >= totalCount) {
+      return;
+    } else {
+      nextBtn.classList.remove('disabled');
+    }
+    const newCount = count + 1;
+    location.hash = '#result' + newCount;
+    currentCount.innerText = newCount;
+    if (newCount >= totalCount) {
+      nextBtn.classList.add('disabled');
+    }
+    if (newCount > 0) {
+      prevBtn.classList.remove('disabled');
+    }
+  } else {
+    if (count <= 0) {
+      return;
+    } else {
+      prevBtn.classList.remove('disabled');
+    }
+    const newCount = count - 1;
+    location.hash = '#result' + newCount;
+    currentCount.innerText = newCount;
+    if (newCount <= 0) {
+      prevBtn.classList.add('disabled');
+    }
+    if (newCount < totalCount) {
+      nextBtn.classList.remove('disabled');
+    }
+  }
+}
+
+function search() {
+  const searchBoxWrapper = document.querySelector('.search-box-wrapper');
+  const searchBox = searchBoxWrapper.querySelector('.searchbox');
+  searchBox.focus();
+  const searchNav = searchBoxWrapper.querySelector('.result-nav');
+  const totalCount = searchNav.querySelector('.total');
+  const currentCount = searchNav.querySelector('.current');
+  const searchGo = searchBoxWrapper.querySelector('#submit-search');
+  const searchQuery = searchBox.value.toLowerCase();
+  if (!searchQuery.length) {
+    clearSearch(searchGo, searchNav, searchBox);
+    return;
+  }
+  const sections = document.querySelectorAll('.section');
+  sections.forEach((section) => {
+    removeHighlights(section);
+    if (section.innerText.toLowerCase().includes(searchQuery)) {
+      section.classList.add('searched');
+      const detailEls = section.querySelectorAll('details');
+      detailEls.forEach((detailEl) => detailEl.setAttribute('open', true));
+      highlightTerm(searchQuery, section);
+    } else {
+      section.classList.remove('searched');
+      const detailEls = section.querySelectorAll('details');
+      detailEls.forEach((detailEl) => detailEl.removeAttribute('open'));
+    }
+  });
+  const results = document.querySelectorAll('.search-term');
+  if (results.length > 0) {
+    results.forEach((result, i) => {
+      const count = i + 1;
+      result.id = 'result' + count;
```

Comment:
I worry about calling this `result` - there's a good chance someone adds some other `result` in the future that conflicts.

Does this have to be an ID, or can you make it a class and use

```
someContainer.querySelector(`.result-${i}`)
```

to find it instead?

Reply:
I used id so I can use location.hash to take the user to the search term